eventbridge-pipes-ddbstream-sfn-terraform/README.md

```diff
@@ -0,0 +1,95 @@
+# DynamoDB Stream to Step Functions with EventBridge Pipes
+
+This pattern shows how to use Amazon EventBridge Pipes to launch a AWS Step Functions state machine with a message coming from an Amazon DynamoDB stream
+
+![Pipes diagram](./ArchDiagram.png)
+
+Learn more about this pattern at Serverless Land Patterns: https://serverlessland.com/patterns/eventbridge-pipes-ddbstream-sfn-terraform
+
+Important: this application uses various AWS services and there are costs associated with these services after the Free Tier usage - please see the [AWS Pricing page](https://aws.amazon.com/pricing/) for details. You are responsible for any AWS costs incurred. No warranty is implied in this example.
+
+* [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) (AWS SAM) installed
+* [Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/aws-get-started) installed
+
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ``` 
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+2. Change directory to the pattern directory:
+    ```
+    cd eventbridge-pipes-ddbstream-sfn-terraform
+    ```
+3. From the command line, initialize terraform to download and install the providers defined in the configuration:
+    ```
+    terraform init
+    ```
+4. From the command line, apply the configuration in the main.tf file:
+    ```
+    terraform apply
+    ```
+5. During the prompts:
+    * Enter yes
+6. Note the outputs from the deployment process. These contain the resource names and/or URLs which are used for testing.
+
+## How it works
+
+Previously, whenever we needed DynamoDB Streams to send messages to Step Functions based on changes made to your DynamoDB table, we had to implement a Lambda function in between DynamoDB and Step Functions because Amazon DynamoDB streams did not support AWS Step Functions as a target directly.
+
+But at re:Invent 2022, AWS launched a new feature for Amazon EventBridge known as EventBridge Pipes.
+
+Now, we can directly integrate DynamoDB streams with AWS Step Functions without the need for having an additional Lambda function.
+
+## Testing
+
+1. Stream logs from the Step Functions log group
+
+```
+sam logs --cw-log-group <LogGroup Name> --tail
```

Comment:
Add note to use SFNLog output in the command.

Reply:
Sorry, I don't understand.  What do you mean?

Reply:
I mean add an instruction to use the value from `SFNLog` in the `sam logs ` command (i.e. replace LogGroup Name with the value of SFNLog).

Reply:
The LogGroup name is dynamically generated using a combination of the Stack name and a randomly generated string. I would like to keep the naming convention generic.